webapp/src/app.tsx

```diff
@@ -2059,18 +2059,29 @@ export class ProjectView
                         });
                     }
                 }
+                pxt.tickEvent("deploy.start")
                 return pxt.commands.deployAsync(resp, {
-                    reportDeviceNotFoundAsync: (docPath, compileResult) => this.showDeviceNotFoundDialogAsync(docPath, compileResult),
-                    reportError: (e) => core.errorNotification(e),
+                    reportDeviceNotFoundAsync: (docPath, compileResult) => {
+                        pxt.tickEvent("deploy.devicenotfound")
+                        return this.showDeviceNotFoundDialogAsync(docPath, compileResult)
+                    },
+                    reportError: (e) => {
+                        pxt.tickEvent("deploy.reporterror")
+                        return core.errorNotification(e)
+                    },
                     showNotification: (msg) => core.infoNotification(msg)
                 })
+                    .then(() => {
+                        pxt.tickEvent("deploy.finished")
```

Comment:
you could actually also store the duration of deploy -- just as a nice side data.